src/controllers/sendEmailController.ts

```diff
@@ -0,0 +1,82 @@
+import shiftRotaService from "../services/shiftRotaService";
+import ShiftRotaRepository from "../repositories/shiftRotaRepository";
+import shiftRotaModel from '../models/shiftRotaModel'
+import sendEmail1 from "../services/sendEmailService";
+
+
+const sendEmailtoAgent= async () => { //email112 powinien przyjmowac jako atrybut shift rota z linijki 12
+
+const shiftRota = new shiftRotaService(new ShiftRotaRepository(shiftRotaModel));
+const todayShifts = await shiftRota.getNextWeekShifts();
+type agent = {
+
+    name: string,
+    date: string,
+    hours: string
+    
+};
+let agents:string[] = [];   // unique list of agents
+let shifts:agent[] = [];
+
+todayShifts.forEach(day1=>{
+    let date1:string= day1.date;
+    let agents1 = day1.agents;
+    let hours1 = day1.hours;
+
+    for(let j =0; j<agents1.length;j++){
+        let name:string = agents1[j];
+        let hours2:string = hours1[j];
```

Comment:
I'd rename hours1, hours2 etc to something more meaningful

additionally to shorten the code you can assign those values to 'shift' object entries in line 37 - no point in creating a variable just to use it in object value assignment

Reply:
I renamed variables to more meaningful names.

For additional point - I left variables as they are, because they are being referred many times